app/src/main/java/com/iti/android/team1/ebuy/ui/category/view/CategoryFragment.kt

```diff
@@ -1,48 +1,159 @@
 package com.iti.android.team1.ebuy.ui.category.view
 
 import android.os.Bundle
+import android.util.Log
 import android.view.*
 import android.widget.Toast
 import androidx.appcompat.widget.SearchView
 import androidx.fragment.app.Fragment
-import androidx.lifecycle.ViewModelProvider
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.lifecycleScope
 import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
 import com.iti.android.team1.ebuy.R
 import com.iti.android.team1.ebuy.databinding.FragmentCategoryBinding
+import com.iti.android.team1.ebuy.model.datasource.repository.Repository
+import com.iti.android.team1.ebuy.model.networkresponse.ResultState
+import com.iti.android.team1.ebuy.model.pojo.Categories
+import com.iti.android.team1.ebuy.model.pojo.Products
 import com.iti.android.team1.ebuy.ui.category.viewmodel.CategoryViewModel
+import com.iti.android.team1.ebuy.ui.category.viewmodel.CategoryViewModelFactory
+import kotlinx.coroutines.flow.buffer
 
 class CategoryFragment : Fragment() {
 
     private var _binding: FragmentCategoryBinding? = null
 
+    private var defaultCategoryId = 395727569125
+    private var defaultProductType = "SHOES"
```

Comment:
Is it required to use these default values? I think using `defaultProductType` is not used out of the scope of the Fab button so you can safely delete it